python/lsst/afw/table/exposure.cc

```diff
@@ -25,23 +25,152 @@
 //#include <pybind11/stl.h>
 
 namespace py = pybind11;
+using namespace pybind11::literals;
+
+#include "lsst/afw/coord/Coord.h"
+#include "lsst/afw/detection/Psf.h"  // forward-declared by Exposure.h
+#include "lsst/afw/geom/Box.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/polygon/Polygon.h"  // forward-declared by Exposure.h
+#include "lsst/afw/image/ApCorrMap.h"  // forward-declared by Exposure.h
+#include "lsst/afw/image/Calib.h"  // forward-declared by Exposure.h
+#include "lsst/afw/image/VisitInfo.h"  // forward-declared by Exposure.h
+#include "lsst/afw/image/Wcs.h"  // forward-declared by Exposure.h
+#include "lsst/afw/table/BaseRecord.h"
+#include "lsst/afw/table/BaseTable.h"
+#include "lsst/afw/table/Catalog.h"
+#include "lsst/afw/table/Exposure.h"
+#include "lsst/afw/table/pybind11/catalog.h"
+#include "lsst/afw/table/pybind11/sortedCatalog.h"
 
 namespace lsst {
 namespace afw {
 namespace table {
 
+typedef py::class_<CatalogT<ExposureRecord>,
+                   std::shared_ptr<CatalogT<ExposureRecord>>> PyBaseExposureCatalog;
+
+typedef py::class_<SortedCatalogT<ExposureRecord>,
+                   std::shared_ptr<SortedCatalogT<ExposureRecord>>,
+                   CatalogT<ExposureRecord>> PySortedBaseExposureCatalog;
+
+typedef py::class_<ExposureCatalogT<ExposureRecord>,
+                   std::shared_ptr<ExposureCatalogT<ExposureRecord>>,
+                   SortedCatalogT<ExposureRecord>> PyExposureCatalog;
+
+/**
+Declare constructors and member and static functions for a pybind11 ExposureCatalog
+
+@tparam RecordT  Record type, e.g. BaseRecord or SimpleRecord.
+
+@param[in] cls  Catalog pybind11 class.
+*/
+void declareExposureCatalog(PyExposureCatalog & cls) {
```

Comment:
But really, why does it exist? It is not templated so can go directly into the module section.